frontend/packages/core/src/AppProvider/index.tsx

```diff
@@ -19,9 +19,14 @@ import { Theme } from "./themes";
 import type { ConfiguredRoute, Workflow, WorkflowConfiguration } from "./workflow";
 import ErrorBoundary from "./workflow";
 
+export interface WorkflowIcon {
+  path: string;
+}
+
 export interface UserConfiguration {
   [packageName: string]: {
-    [key: string]: ConfiguredRoute;
+    icon: WorkflowIcon;
+    [key: string]: WorkflowIcon | ConfiguredRoute;
```

Comment:
curious, why does WorkflowIcon need to be set in 2 places rather than just `icon` or vice versa?

Reply:
Yeah this one was weird. It's because we're defining `[key: string]` as an unknown property, which technically will match on `icon` as well. So it has to be OR'ed into the typing to alleviate any potential errors. We could technically remove the `icon` property and just look into the workflow configuration for it, but that seemed counter-intuitive when we know what it is.